File: VB-Examples/07_DocumentOperation/RecurseAllDocumentObject.vb
Imports System.IO
Imports System.Text
Imports Spire.Doc
Imports Spire.Doc.Documents

Namespace RecurseAllDocumentObject
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new Document object
			Dim document As New Document()

			' Load the document from the specified file
			document.LoadFromFile("..\..\..\..\..\..\Data\Sample.docx")

			' Create a new StringBuilder to store the content
			Dim builder As New StringBuilder()

			' Iterate through each Section in the document
			For Each section As Section In document.Sections
				' Get the index of the current section
				Dim SectionIndex As Integer = document.GetIndex(section)
				builder.AppendLine(String.Format("section index {0} has following ChildObjects", SectionIndex))

				' Iterate through each ChildObject in the section's body
				For Each obj As DocumentObject In section.Body.ChildObjects
					' Append the index and type of the ChildObject to the StringBuilder
					builder.AppendLine(String.Format("Index : {0}, ChildObject Type: {1}", section.Body.GetIndex(obj), obj.DocumentObjectType))
					
					' Check if the ChildObject is a Paragraph
					If obj.DocumentObjectType.Equals(DocumentObjectType.Paragraph) Then
						Dim paragraph As Paragraph = TryCast(obj, Paragraph)
						' Append the index and type of the Paragraph and its ChildObjects to the StringBuilder with indentation
						builder.AppendLine(String.Format(vbTab & "Paragraph index {0} has following ChildObjects", section.Body.GetIndex(paragraph)))
						For Each obj2 As DocumentObject In paragraph.ChildObjects
							builder.AppendLine(String.Format(vbTab & "Index : {0}, ChildObject Type: {1}", paragraph.GetIndex(obj2), obj2.DocumentObjectType))
						Next obj2
					End If
				Next obj
				builder.AppendLine(" ")
			Next section

			' Write the content of the StringBuilder to a text file
			File.WriteAllText("RecurseAllDocumentObject.txt", builder.ToString())

			' Release all resources used by the Document object
			document.Dispose()

			'Launching the Word file.
			TextViewer("RecurseAllDocumentObject.txt")


		End Sub

		Private Sub TextViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace

File: CS-Examples/07_DocumentOperation/RecurseAllDocumentObject.cs
using System;
using System.IO;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;

namespace RecurseAllDocumentObject
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new instance of the Document class
			Document document = new Document();

			// Load the document from the specified file path using a relative path
			document.LoadFromFile(@"..\..\..\..\..\..\Data\Sample.docx");

			// Create a StringBuilder object to store the output string
			StringBuilder builder = new StringBuilder();

			// Iterate through each section in the document
			foreach (Section section in document.Sections)
			{
				// Get the index of the current section
				int sectionIndex = document.GetIndex(section);
				
				// Append a formatted string indicating the section index and its child objects
				builder.AppendLine(string.Format("section index {0} has following ChildObjects", sectionIndex));

				// Iterate through each child object in the section's body
				foreach (DocumentObject obj in section.Body.ChildObjects)
				{
					// Get the index and type of the current child object
					builder.AppendLine(string.Format("Index: {0}, ChildObject Type: {1}", section.Body.GetIndex(obj), obj.DocumentObjectType));
					
					// Check if the child object is a paragraph
					if (obj.DocumentObjectType.Equals(DocumentObjectType.Paragraph))
					{
						// Convert the child object to a Paragraph
						Paragraph paragraph = obj as Paragraph;
						
						// Append a formatted string indicating the paragraph index and its child objects
						builder.AppendLine(string.Format("\tParagraph index {0} has following ChildObjects", section.Body.GetIndex(paragraph)));
						
						// Iterate through each child object in the paragraph
						foreach (DocumentObject obj2 in paragraph.ChildObjects)
						{
							// Append a formatted string indicating the index and type of the child object
							builder.AppendLine(string.Format("\tIndex: {0}, ChildObject Type: {1}", paragraph.GetIndex(obj2), obj2.DocumentObjectType));
						}
					}
				}
				
				// Append a blank line to separate sections
				builder.AppendLine(" ");
			}

			// Write the contents of the StringBuilder to a text file
			File.WriteAllText("RecurseAllDocumentObject.txt", builder.ToString());

			// Dispose of the document object to free up resources
			document.Dispose();

            //Launching the Word file.
            TextViewer("RecurseAllDocumentObject.txt");


        }

        private void TextViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}
